Create usercripts-api.md (#711)
AdguardTeam/KnowledgeBase: docs/adguard-browser-extension/user-scripts-api.md

---
title: User Scripts API
sidebar_position: 8
---

:::info

This article is about AdGuard Browser Extension, which safeguards only your browser. To protect your entire device, [download the AdGuard app](https://adguard.com/download.html?auto=true).

:::

When adding a custom filter or user rule, a notification will appear asking you to enable developer mode to use these features in the extension:

![Notification *border](https://cdn.adtidy.org/content/Kb/ad_blocker/browser_extension/dev-mode-ext.png)

This is because, starting with version 5.2, AdGuard Browser Extension uses the User Scripts API . In other words, the extension now uses script-based rules.

## What is the User Scripts API?

> The User Scripts API is a set of browser-provided functions and capabilities that allow extensions to run custom JavaScript scripts on web pages. These are often used to modify or enhance websites in ways that aren’t originally provided by the website itself.

Reviewer: > The User Scripts API is a set of browser-provided functions and capabilities that allow extensions to run custom JavaScript scripts on web pages

can it be simpler?
> The User Scripts API is a tool that allows extensions to run customized JavaScript scripts on webpages.

---
title: User Scripts API
sidebar_position: 8
---

:::info

This article is about AdGuard Browser Extension, which safeguards only your browser. To protect your entire device, [download the AdGuard app](https://adguard.com/download.html?auto=true).

:::

> When adding a custom filter or user rule, a notification will appear asking you to enable developer mode to use these features in the extension:

Reviewer 2: ```suggestion
When adding a custom filter or user rule, a notification will appear asking you to enable the Developer mode to use these features in the extension.
```

![Notification *border](https://cdn.adtidy.org/content/Kb/ad_blocker/browser_extension/notification-allow-user-scripts.jpg)

> Starting with version 5.2, AdGuard Browser Extension uses the User Scripts API. Chrome requires this new API to follow best practices for user safety.

Reviewer 3: я бы тут где-то линканул нашу историю про то как мы ревью проходили и тд

> This is because, starting with version 5.2, AdGuard Browser Extension uses the User Scripts API. This new API is required by Chrome to follow the best practices for user safety.

Reviewer 2: ```suggestion
Starting with version 5.2, AdGuard Browser Extension uses the User Scripts API. Chrome requires this new API to follow best practices for user safety.
```

## What is the User Scripts API?

The User Scripts API is a tool that allows extensions to run customized JavaScript scripts on webpages. These are often used to modify or enhance websites in ways that aren’t originally provided by the website itself.

With the shift to Chrome’s Manifest V3, many older methods of injecting scripts are limited or delayed. However, there is an exception for the User Scripts API.

## How to enable the Developer mode

> To use custom filters and JavaScript rules in the tab *User Rules*, you have to set up your browser first. This will give the extension a reliable and secure method of injecting scripts into web pages at the optimal time. To do so, follow the instruction below, depending on your Chrome version.

Reviewer 2: ```suggestion
To use custom filters and JavaScript rules in the *User Rules* tab, you must set up your browser first. This will give the extension a reliable and secure method of injecting scripts into webpages at the optimal time. Follow the instructions below depending on your Chrome version.
```

### For Chrome versions prior to 138

> You must enable the Developer mode for extensions. To do it, open the Chrome menu by clicking the three dots in the upper-right corner, go to *Extensions* ⭢ *Manage Exetensions*. Toggle the "Developer mode" switch in the upper-right corner:

Reviewer 2: ```suggestion
You must enable Developer mode for extensions. To do so, open the Chrome menu by clicking the three dots in the upper-right corner, go to *Extensions* ⭢ *Manage extensions*, and toggle the *Developer mode* switch.
```

![Manage Extensions *border](https://cdn.adtidy.org/content/Kb/ad_blocker/browser_extension/developer-mode-enable1.jpg)

### For Chrome version 138 and later

> You must enable the *Allow User Scripts* toggle. To do it, open the Chrome menu by clicking the three dots in the upper-right corner, go to *Extensions* ⭢ *Manage Exetensions*. Toggle the *Allow User Scripts* switch.

Reviewer 2: ```suggestion
You must enable the *Allow User Scripts* toggle. To do so, open the Chrome menu by clicking the three dots in the upper-right corner, go to *Extensions* ⭢ *Manage extensions*, and toggle the *Allow User Scripts* switch.
```
